paas_app_charmer/framework.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Framework related base classes."""
+import typing
+
+import pydantic
+
+
+class FrameworkConfig(pydantic.BaseModel):
+    """Base class for framework config models."""
+
+    @pydantic.model_validator(mode="before")
+    @classmethod
+    def secret_key_id(cls, data: dict[str, str | int | bool | dict[str, str] | None]) -> dict:
+        """Read the *-secret-key-id style configuration.
+
+        Args:
+            data: model input.
+
+        Returns:
+            modified input with *-secret-key replaced by the secret content of *-secret-key-id.
+
+        Raises:
+            ValueError: if the *-secret-key-id is invalid.
+        """
+        secret_key_field = "secret_key"
+        if secret_key_field not in cls.model_fields:
+            secret_key_field = "app_secret_key"
+        secret_key_config_name = cls.model_fields[secret_key_field].alias
+        assert secret_key_config_name
```

Comment:
Should we convert this to a check and raising a ValueError? Or is this just for typing?

Reply:
Because of Pydantic, we need to use "reflect" to create a common function for all four framework configurations. Since we are using "reflect", there's no protection against issues like an attribute name mismatch. Therefore, the assertion here guards against that situation, i.e., the programmer made a mistake in `paas-app-charmer` rather than incorrect input values during runtime.

Reply:
I agree with the check, just saying that perhaps it would be better to raise a value error instead of using an assert?

Reply:
Let's switch to NotImplementedError to make it easier to identify the issue

Reply:
Updated, thanks!